tests/unit/test_data.py

```diff
@@ -9,3 +14,35 @@ def test_repeating_loader():
         assert next(loader) == 1
         assert next(loader) == 2
         assert next(loader) == 3
+
+
+def test_dataloader_drop_last(tmpdir):
```

Comment:
I think the coverage should be improved using [parameterization](https://github.com/microsoft/DeepSpeed/blob/336dd089e5e11d46daf3abfad048d71b72912504/tests/unit/test_fp16.py#L609). In particular, I think at least these four cases should be covered where drop_last=[true,False] and total_samples is (not) divisible by train_batch_size. For example, assuming total_samples=50:

1. drop_last=True, train_batch_size=1
2. drop_last=True, train_batch_size=4
3. drop_last=False, train_batch_size=1
4. drop_last=False, train_batch_size=4

There might be other corner cases (e.g., world_size=2), but I think those 4 are required. Parametrization is heavily used in other unit tests, in case you are not familiar with it. Thanks.

Reply:
Sure I will add that too. Thanks!

Reply:
Done. Can you please review @tjruwase Thanks!